politeiad/backendv2/tstorebe/plugins/pi/cmds.go

```diff
@@ -88,30 +80,54 @@ func (p *piPlugin) cmdSetBillingStatus(token []byte, payload string) (string, er
 		}
 	}
 
-	// Ensure no billing status already exists
 	bscs, err := p.billingStatusChanges(token)
 	if err != nil {
 		return "", err
 	}
-	if len(bscs) > 0 {
-		return "", backend.PluginError{
-			PluginID:     pi.PluginID,
-			ErrorCode:    uint32(pi.ErrorCodeBillingStatusChangeNotAllowed),
-			ErrorContext: "can not set billing status more than once",
+	if len(bscs) == 0 {
+		// If no billing status was set yet, verify the following:
+		// 1. The new billing status is not `active`.
+		// 2. Proposal's vote ended and it was approved.
+		if sbs.Status == pi.BillingStatusActive {
+			return "", backend.PluginError{
+				PluginID:  pi.PluginID,
+				ErrorCode: uint32(pi.ErrorCodeBillingStatusChangeNotAllowed),
+				ErrorContext: "can not set the billing status of an active " +
+					"proposal to 'active'",
+			}
 		}
-	}
-
-	// Ensure record's vote ended and it was approved
-	vsr, err := p.voteSummary(token)
-	if err != nil {
-		return "", err
-	}
-	if vsr.Status != ticketvote.VoteStatusApproved {
-		return "", backend.PluginError{
-			PluginID:  pi.PluginID,
-			ErrorCode: uint32(pi.ErrorCodeBillingStatusChangeNotAllowed),
-			ErrorContext: "setting billing status is allowed only if " +
-				"proposal vote was approved",
+		vsr, err := p.voteSummary(token)
+		if err != nil {
+			return "", err
+		}
+		if vsr.Status != ticketvote.VoteStatusApproved {
+			return "", backend.PluginError{
+				PluginID:  pi.PluginID,
+				ErrorCode: uint32(pi.ErrorCodeBillingStatusChangeNotAllowed),
+				ErrorContext: "setting billing status is allowed only if " +
+					"proposal vote was approved",
+			}
+		}
+	} else {
+		// If a billing status was already set, a new billing status change is
+		// allowed only if the number of all billing status changes does not exceed
+		// the 'allowedbillingstatuschangesmax' plugin setting.
+		if uint32(len(bscs))+1 > p.allowedBillingStatusChangesMax {
```

Comment:
Shouldn't this validation also be pulled out of this if-else statement and done prior to validating the billing status transition?

What if the sysadmin sets the max billing status changes to 0?  The software would not work as expected since it would allow the first billing status change.